app/(dashboard)/edit-item/[id]/page.tsx

```diff
@@ -0,0 +1,255 @@
+'use client';
+
+import ButtonRounded from '@/components/buttons/ButtonRounded';
+import { useForm } from 'react-hook-form';
+import { PartialItem } from '@/types/supabaseTypes';
+import UploadImageInput from '@/components/form/UploadImageInput';
+import { createClientComponentClient } from '@supabase/auth-helpers-nextjs';
+import { useEffect, useState } from 'react';
+import { useRouter } from 'next/navigation';
+import editRow from '@/supabase/models/editRow';
+
+const EditItemPage = ({ params }: { params: { id: number } }) => {
+  const [imgSrc, setImageSrc] = useState('');
+  const [generalError, setGeneralError] = useState('');
+  const router = useRouter();
+
+  const {
+    register,
+    handleSubmit,
+    watch,
+    reset,
+    formState: { errors },
+  } = useForm({
+    defaultValues: {
+      item_name: '',
+      item_description: '',
+      postcode: '',
+      condition: '',
+      item_type: '',
+      size: '',
+      item_subtype: '',
+      postable: false,
+      collectible: false,
+      postage_covered: false,
+    },
+  });
+
+  useEffect(() => {
+    const fetchItemData = async () => {
+      const supabase = createClientComponentClient();
+
+      const { data: itemData } = await supabase
+        .from('items')
+        .select('*,profiles(*)')
+        .eq('id', params.id)
+        .single();
+
+      reset({
+        item_name: itemData.item_name,
+        item_description: itemData.item_description,
+        postcode: itemData.postcode,
+        condition: itemData.condition,
+        item_type: itemData.item_type,
+        size: itemData.size,
+        item_subtype: itemData.item_subtype,
+        postable: itemData.postable,
+        collectible: itemData.collectible,
+        postage_covered: itemData.postage_covered,
+      });
+
+      setImageSrc(itemData.imageSrc);
+    };
+    fetchItemData();
+  }, []);
+
+  const onSubmit = async (data: PartialItem) => {
+    const itemData: PartialItem = {
+      id: params.id,
+      imageSrc: imgSrc,
+      ...data,
+    };
+    try {
+      await editRow('items', itemData, 'id', itemData.id);
```

Comment:
I wonder if instead of using this editRow function (which, on a personal note, I'm not sure I love) we could just assemble the item and do an `upsert` to supabase.

That might allow us to just have the one route cover both use cases ("edit-item" would still make sense if we are editing a new item).

I'm throwing this idea about but it might not be all that good 😅 - while we mull over the `useRoute` issue it might be worth looking at though